test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go

```diff
@@ -52,30 +76,436 @@ var _ = Describe("[performance] Cgroups and affinity", Ordered, func() {
+				}
+			},
+			Entry("[test_id:68385] Verify ovs cgroup with crun ", true, true, true, true),
+			Entry("[test_id:64098] Verify ovs cgroup with runc", false, true, true, true),
+		)
+	})
+
+	Describe("Affinity", func() {
+		Context("ovn-kubenode Pods affinity ", func() {
+			testutils.CustomBeforeAll(func() {
+				initialProfile = profile.DeepCopy()
+			})
+			It("[test_id:64100] matches with ovs process affinity", func() {
+				ovnPod, err := getOvnPod(workerRTNode)
+				Expect(err).ToNot(HaveOccurred(), "Unable to get ovnPod")
+
+				ovnContainersids, err := getOvnPodContainers(&ovnPod)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Generally there are many containers inside a kubenode pods
+				// we don't need to check cpus used by all the containers
+				// we take first container
+				cpus := getCpusUsedByOvnContainer(workerRTNode, ovnContainersids[0])
+				testlog.Infof("Cpus used by ovn Containers are %s", cpus)
+				pidList, err := getOVSServicesPid(workerRTNode)
+				Expect(err).ToNot(HaveOccurred())
+
+				for _, pid := range pidList {
+					cmd := []string{"taskset", "-pc", pid}
+					cpumask, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
+					Expect(err).ToNot(HaveOccurred(), "failed to fetch cpus of %s", pid)
+					mask := strings.SplitAfter(cpumask, " ")
+					maskSet, err := cpuset.Parse(mask[len(mask)-1])
+					testlog.Infof("cpus used by ovs services are: %s", maskSet.String())
+					Expect(cpus).To(Equal(maskSet.String()), "affinity of ovn kube node pods(%s) do not match with ovservices(%s)", cpus, maskSet.String())
+					if err != nil {
+						testlog.Error(err.Error())
+					}
+				}
+			})
+
+			It("[test_id:64101] Creating gu pods modifies affinity of ovs", func() {
+				var testpod *corev1.Pod
+				var err error
+				testpod = pods.GetTestPod()
+				testpod.Namespace = testutils.NamespaceTesting
+				testpod.Spec.Containers[0].Resources = corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("2"),
+						corev1.ResourceMemory: resource.MustParse("200Mi"),
+					},
+				}
+				testpod.Spec.NodeSelector = map[string]string{testutils.LabelHostname: workerRTNode.Name}
+				err = testclient.Client.Create(context.TODO(), testpod)
+				Expect(err).ToNot(HaveOccurred())
+				testpod, err = pods.WaitForCondition(client.ObjectKeyFromObject(testpod), corev1.PodConditionType(corev1.PodReady), corev1.ConditionTrue, 5*time.Minute)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testpod.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed))
+
+				cmd := []string{"taskset", "-pc", "1"}
+				testpodCpus, err := pods.ExecCommandOnPod(testclient.K8sClient, testpod, cmd)
+				testlog.Infof("%v pod is using %v cpus", testpod.Name, string(testpodCpus))
+
+				By("Get ovnpods running on the worker cnf node")
+				ovnPod, err := getOvnPod(workerRTNode)
+				Expect(err).ToNot(HaveOccurred(), "Unable to get ovnPod")
+
+				By("Get cpu used by ovn pod containers")
```

Comment:
There is a race condition here. CPU manager only runs the reconcile loop every 5 seconds or so.

Reply:
@MarSik this is the flow of the test case 

1.  Create a gu testpod (no annotations are used)
2.  Get cpu used  this testpod (using taskset)
3.  Find the cgroup used by one of the containers of ovn-kubenode pod,
4. There is a  wait of 30seconds here 
5. check it's cpuset.cpus 
6. Verify the cpuset.cpus match with the ovs-vswitchd 

Can you let me know where i can hit the race condition

Reply:
Where is the 30 second delay? The race is between the cpus being allocated to the gu pod in 1 and the cpu manager updating the cpu mask for ovnkube-node (should be about 5 seconds).

Reply:
@MarSik the 30 second delay is in the https://github.com/openshift/cluster-node-tuning-operator/pull/746/files#diff-ce7d5cbe8bfc6cc4efbf81602d5f238b37bafcbb68bf2c1ad66f1e1b148a33c5R580